core-bundle/src/Cron/CronJob.php

```diff
@@ -29,13 +29,29 @@ public function __construct(private object $service, private string $interval, p
         }
     }
 
-    public function __invoke(string $scope): void
+    public function __invoke(string $scope): ProcessCollection|null
     {
         if (\is_callable($this->service)) {
-            ($this->service)($scope);
-        } else {
-            $this->service->{$this->method}($scope);
+            return ($this->service)($scope);
         }
+
+        return $this->service->{$this->method}($scope);
+    }
+
+    public function isAsync(): bool
+    {
+        if (\is_callable($this->service)) {
+            return false;
+        }
+
+        $ref = new \ReflectionMethod($this->getService(), $this->getMethod());
+        $returnType = $ref->getReturnType();
+
+        if (!$returnType instanceof \ReflectionNamedType) {
+            return false;
+        }
+
+        return ProcessCollection::class === $returnType->getName();
```

Comment:
Rather than using reflection - should we may be introduce an interface instead? That way `CronJob::isAsync` could simply return `$this->service instanceof AsyncCronJobInterface` - or the interface itself could mandate an `isAsync` method that the service has to implement. Just a thought.

Reply:
How about adding `isAsync` to the `AsCronJob` attribute and the associated tag?

Reply:
I would probably indeed add an interface. You need to ensure the correct return type. Just adding the attribute doesn't fulfil any contract.

Reply:
I'll give the interface variant a try - let's see how we like that :)

Reply:
Though I get why the reflection is used - since it ultimately depends on what the CronJob's method is ultimately returning.

Reply:
I think controlling this via an interface would be a confusing mixture of configuration. Everything else is configured by the `contao.cronjob` tag, so why not configure `isAsync` there too?

Reply:
Agreed. It would be weird if you put your `AsCronJob` attribute on a method - but then were forced to use a different method.